client/extensions/woocommerce/state/ui/products/api-plan/actions.js

```diff
@@ -0,0 +1,71 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_EDIT_PRODUCT_APIPLAN_CREATE,
+	WOOCOMMERCE_EDIT_PRODUCT_APIPLAN_CLEAR,
+	WOOCOMMERCE_EDIT_PRODUCT_APIPLAN_STEP_START,
+	WOOCOMMERCE_EDIT_PRODUCT_APIPLAN_STEP_END,
+} from '../../../action-types';
+
+/* TODO: Create the plan based on our current edits.
+export function createApiPlanForEdits( productCategoryEdit, productEdits, variationEdits ) {
+	const plan = [];
+
+	// TODO: sequentially go through edit state and create steps.
+	...categories.creates
+	...categories.updates
+	...categories.deletes
+
+	...product.creates
+	...product.updates
+	...product.deletes
+
+	...variation.creates
+	...variation.updates
+	...variation.deletes
+
+	return createApiPlan( plan );
+}
+*/
+
+/* TODO: Figure out the next step and start it.
+export function continueApiPlan( plan ) {
+	//const nextStep = getNextStep( plan );
+
+	//...api call for operation
+
+	//return startApiPlanStep( nextStep );
+}
+*/
+
+export function clearApiPlan() {
+	return {
+		type: WOOCOMMERCE_EDIT_PRODUCT_APIPLAN_CLEAR,
+	};
+}
+
+// The functions below are used internally and for testing.
```

Comment:
I just tried this and if I add a jsdoc block (even with @private) it wants params and returns on it. These things are very likely to change, so I'd like to just leave this off for now.